src/MongoDB.Bson/Serialization/Serializers/ObjectSerializer.cs

```diff
@@ -66,6 +78,17 @@ public static ObjectSerializer Instance
             get { return __instance; }
         }
 
+        // public properties
+        /// <summary>
+        /// Gets the discriminator convention.
+        /// </summary>
+        public IDiscriminatorConvention DiscriminatorConvention => _discriminatorConvention;
```

Comment:
As a general rule serializers that can be configured should have public properties allowing the configuration to interrogated.

These should be public properties.

See for example `Int32Serializer` that has public `Converter` and `Representation` properties.

Reply:
>As a general rule serializer

I can argue with this. `Int32Serializer` has `Converter` and `Representation` fields because these fields are part of the interfaces which this serializer implements. And the implemented fields are used in the driver (not only in tests). So, it's completely clear why they are public.

At the same time, the fields which you made public are not a part of any interfaces and not used in any driver logic. So, I don't see why somebody will need to have access to these fields.

Also, I think we have to be careful with parts of the driver which we're going to make public since we cannot remove(or change) them later because it will be a breaking change. So, I would say that one of the general rules should be we should not make anything public until this functionality is really required for users

Reply:
Keeping things private is a good general principle. I think applying it here is carrying it to an unnecessary extreme. Nevertheless I will remove these two public properties because it is more important to progress this ticket to an LGTM.